pkgs/applications/misc/owmods-cli/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, pkg-config
+, openssl
+, libsoup
+, fetchFromGitHub
+, installShellFiles
+, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "owmods-cli";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "Bwc9876";
+    repo = "ow-mod-man";
+    rev = "cli_v${version}";
+    sparseCheckout = [
+      "owmods_cli"
+      "owmods_core"
+      "xtask"
+    ];
```

Comment:
So, something unfortunate happened here. I'm not a Nix expert, but I have accidentally caused this one before, so I'm aware of how it happened.

When I first tried to evaluate this derivation, it failed because the SHA256 sum did not match. When I updated it, the build failed because files that were needed were missing. I removed the `sparseCheckout` line and things built just fine.

The way Nix handles caching, I believe it computes a name + hash and then if that exists in the store, it won't even check to see if it's valid; it can't really anyway, without defeating the purpose of hashing. So it's very possible to accidentally use an old hash after changing a field and wrongly think that it worked, because whatever you changed did not wind up changing the name.

In the future, whenever you change things that should change the hash, avoid this by replacing the hash with `lib.fakeHash` or `lib.fakeSha256`, like this:
```nix
    sha256 = lib.fakeHash;
```
then, when Nix runs, it will tell you the correct hash, which you can then paste into your derivation.

But, in this case, the `sparseCheckout` just doesn't work, because somehow the CLI of owmods seems to require bits from the GUI. So in this case, removing `sparseCheckout` should be enough, and the hash will match up in that case.

Reply:
Fixed the `sparseCheckout` issue. Removing it really allowed it to work with the old hash.